llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1467,14 +1467,20 @@ static void computeKnownBitsFromOperator(const Operator *I,
     break;
   }
   case Instruction::Call:
-  case Instruction::Invoke:
+  case Instruction::Invoke: {
     // If range metadata is attached to this call, set known bits from that,
     // and then intersect with known bits based on other properties of the
     // function.
     if (MDNode *MD =
             Q.IIQ.getMetadata(cast<Instruction>(I), LLVMContext::MD_range))
       computeKnownBitsFromRangeMetadata(*MD, Known);
-    if (const Value *RV = cast<CallBase>(I)->getReturnedArgOperand()) {
+
+    const CallBase *CB = cast<CallBase>(I);
+
+    if (std::optional<ConstantRange> Range = CB->getRange())
+      Known = Range->toKnownBits();
```

Comment:
Think this should be `Known.unionWith(Range->toKnownBits())`.

Reply:
hmm maybe it was suggested to forbid range metadata on calls in the future https://github.com/llvm/llvm-project/pull/84627#discussion_r1518589822 but maybe needed until then or at least swap the order so this is done before the range metadata handling

Reply:
fixed now